scripts/Package_BGA/bga.py

```diff
@@ -149,11 +141,10 @@ def crtYdRound(x):
                          shape=padShape,
                          at=[xPadLeft + (col-1) * pitch, yPadTop + rowNum * pitch],
                          size=[padDiameter, padDiameter],
-                         layers=Pad.LAYERS_SMT,
-                         solder_paste_margin_ratio=pasteRatio))
-
-    f.setTags("BGA {} {}".format(balls, pitch))
+                         layers=Pad.LAYERS_SMT))
 
+    f.setTags("{} {} {}".format('CSP' if 'BGA' not in footprint_name and 'CSP' in footprint_name else 'BGA', balls, pitch))
```

Comment:
This is the second time you wrote `'CSP' if 'BGA' not in footprint_name and 'CSP' in footprint_name else 'BGA'`.  I strongly suggest you only do that once and store the resulting string to a variable.